core/src/main/java/com/uber/m3/util/Duration.java

```diff
@@ -193,13 +198,14 @@ public String toString() {
 
         boolean isNegative = nanos < 0;
 
-        long localNanos = Math.abs(nanos);
+        long nanosLocal = Math.abs(nanos);
 
-        long hours = localNanos / NANOS_PER_HOUR;
-        int minutes = (int) ((localNanos % NANOS_PER_HOUR) / NANOS_PER_MINUTE);
-        float seconds = (float) (localNanos % NANOS_PER_MINUTE) / (float) NANOS_PER_SECOND;
+        long hours = nanosLocal / NANOS_PER_HOUR;
+        int minutes = (int) ((nanosLocal % NANOS_PER_HOUR) / NANOS_PER_MINUTE);
+        long secondsInNanos = nanosLocal % NANOS_PER_MINUTE;
+        int nanoOffset = (int) (nanosLocal % NANOS_PER_SECOND);
 
-        StringBuilder buf = new StringBuilder(16);
+        StringBuilder buf = new StringBuilder(24);
```

Comment:
where does 24 come from?

Reply:
This should actually be 25, which is the max possible `toString` return String length ("-2562047h47m16.854775808s" when nanos is `Long.MIN_VALUE`). Will update this and put in a comment.